src/lib/es5.d.ts

```diff
@@ -225,19 +225,37 @@ interface ObjectConstructor {
      * Returns true if existing property attributes cannot be modified in an object and new properties cannot be added to the object.
      * @param o Object to test.
      */
-    isSealed(o: any): boolean;
+    isSealed(o: object): boolean;
+
+    /**
+     * Returns true if existing property attributes cannot be modified in an object and new properties cannot be added to the object.
+     * @param o Object to test.
+     */
+    isSealed(o: any): true;
+
+    /**
+     * Returns true if existing property attributes and values cannot be modified in an object, and new properties cannot be added to the object.
+     * @param o Object to test.
+     */
+    isFrozen(o: object): boolean;
 
     /**
      * Returns true if existing property attributes and values cannot be modified in an object, and new properties cannot be added to the object.
      * @param o Object to test.
      */
-    isFrozen(o: any): boolean;
+    isFrozen(o: any): true;
 
     /**
      * Returns a value that indicates whether new properties can be added to an object.
      * @param o Object to test.
      */
-    isExtensible(o: any): boolean;
+    isExtensible(o: object): boolean;
+
+    /**
+     * Returns a value that indicates whether new properties can be added to an object.
+     * @param o Object to test.
+     */
+    isExtensible(o: any): false;

```

Comment:
These changes are not actually correct, as `object` is a subtype of `any`, and thus, a variable that is a union of `object` and non‑`object` will cause the second overload to be selected.